### PR TITLE
search icon completion

### PR DESCRIPTION
**Pull request by Yanpas**
peek 2017-09-03 18-18
Fallback search.svg belongs to KDE Breeze theme.

### CODE REVIEW

**Comment by Yanpas** [comment]
Will commit check for parents in click slot

**Comment by Yanpas** [comment]
And still I think it's a wrong way to use delegate. Maybe we should stick to `setIndexWidget`?

**Comment by trollixx** [comment]
I've done some digging, and it seems like the right solution would be catching mouse events in reimplemented [QAbstractItemDelegate::editorEvent](http://doc.qt.io/qt-5/qabstractitemdelegate.html#editorEvent).

Here's the debug output I get by just logging `event` parameter:

```
QMouseEvent(MouseButtonPress, LeftButton, localPos=202,7, screenPos=990,431)
QMouseEvent(MouseButtonRelease, LeftButton, localPos=202,7, screenPos=990,431)
QMouseEvent(MouseButtonPress, LeftButton, localPos=200,10, screenPos=988,434)
QMouseEvent(MouseButtonRelease, LeftButton, localPos=200,10, screenPos=988,434)
```

**Comment by trollixx** [comment]
@Yanpas Are you still interested in getting this in? I think the `editorEvent` is the way to go.

**Comment by Yanpas** [comment]
Yep. I've forgot of opened PR, I will send updates this week I think

File: src/libs/ui/mainwindow.cpp [hunk 1]
```diff
@@ -285,22 +285,24 @@ MainWindow::MainWindow(Core::Application *app, QWidget *parent) :
...
-    ui->tocListView->setItemDelegate(new SearchItemDelegate(ui->tocListView));
+    ui->tocListView->setItemDelegate(new SearchItemDelegate(*ui->tocListView, ui->tocListView));
     connect(ui->tocSplitter, &QSplitter::splitterMoved, this, [this]() {
         m_settings->tocSplitterState = ui->tocSplitter->saveState();
     });
 
     createTab();
 
-    connect(ui->treeView, &QTreeView::clicked, this, &MainWindow::openDocset);
-    connect(ui->tocListView, &QListView::clicked, this, &MainWindow::openDocset);
```
Comment (Yanpas) [inline src/libs/ui/mainwindow.cpp]: The focus from search icon was cancelled by this actions

**Comment by Yanpas** [comment]
I've updated the branch

**Comment by trollixx** [comment]
Wow, ten files changed for this tiny feature. I think implementation can be leaner. I'll go over the change in a bit in more detail. Thanks!

**Pull request closed without merging**